docs/CHANGELOG.rst

```diff
@@ -2,6 +2,12 @@
 Changelog
 =========
 
+new_version
+-----------
+
+* Simplified CI testing
```

Comment:
I don't like to call it "simplified". It is not simplified, it is different. The setup proposed here has its good and its bads as the previous one. I don't remember well but I was keeping python version separated because of `tox-conda`. Does it work now? If yes, well we can fo for this "matrix based version". But again, I don't like it that much. Is a bit of spider-sense. I prefer having python versions separated.

Reply:
True! 

`tox-conda` seems to work. At least the tests are not failing. The versions are still separated as defined in the Github action but not in the tox file. As you said it is more a design decision, but currently it is less code and it seems to work.